File: src/app/model/attribute/team-paint.ts
import { AttributeType } from './attribute';
import { BinaryReader } from '../../parser/binary-reader';

export class TeamPaint {
  team: number;
  primaryColor: number;
  accentColor: number;
  primaryFinish: number;
  accentFinish: number;

  static deserialize(br: BinaryReader): TeamPaint {
    const t = new TeamPaint();
    t.team = br.readByte();
    t.primaryColor = br.readByte();
    t.accentColor = br.readByte();
    t.primaryFinish = br.readUInt32();
    t.accentFinish = br.readUInt32();
    return t;
  }
}

export const AttributeTypeTeamPaint: AttributeType = {
  deserialize: (br: BinaryReader): TeamPaint => {
    return TeamPaint.deserialize(br);
  }
};

File: boxcars-wasm/src/model/team_paint.rs
use serde::Serialize;

#[derive(Serialize, Debug)]
pub struct TeamPaint {
    pub primary_color: u8,
    pub accent_color: u8,
    pub primary_finish: u32,
    pub accent_finish: u32,
}

impl TeamPaint {
    pub fn from(tp: &boxcars::attributes::TeamPaint) -> TeamPaint {
        TeamPaint {
            primary_color: tp.primary_color,
            accent_color: tp.accent_color,
            primary_finish: tp.primary_finish,
            accent_finish: tp.accent_finish,
        }
    }
}
